src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -69,7 +94,10 @@ case object ThurloeDatabaseConnector extends DataAccess with LazyLogging {
 
   def lookupIncludingDatabaseId(userId: String, key: String): Future[UserKeyValuePairWithId] =
     for {
-      results <- lookupWithConstraint(x => x.key === key && x.userId === userId)
+      samUser <- lookupSamUser(userId)
+      results <- lookupWithConstraint(x =>
+        x.key === key && (x.userId === samUser.getId || x.userId === samUser.getGoogleSubjectId || x.userId === samUser.getAzureB2CId)
```

Comment:
This could theoretically return multiple records, but I think that would only happen in the case of random id collision?

Reply:
I have to add something to coalesce records dont I?